x-pack/docs/en/security/authentication/user-profile.asciidoc

```diff
@@ -0,0 +1,85 @@
+[role="xpack"]
+[[user-profile]]
+=== User profiles
+
+beta::[]
+
+User profiles provide persistent and stable representations of end-users.
+Persistent means it exists without the need for the user to be online.
+Stable means it has an unique identifier that does not change through
+the lifetime of a deployment and can be used to refer to the associated
+user.
+Being able to uniquely refer to users is a critical function that underpins
+important features like personalisation and collaboration in {kib}.
+
+==== How user profile works
+
+In the {stack}, a user is required to authenticate each request. Each user is
+represented as an object that encapsulates its identity and privileges.
+This object is ephemeral especially for users authenticating from a realm
+backed by external identity providers, e.g. <<saml-realm,SAML>>.
+There is simply no way to refer to a user without that user being online
+at the moment.
+
+User profiles solve the problem by creating profile documents for
+authenticated users and persisting these documents in the profile
+<<system-indices,system index>>. Each profile document is assigned a unique ID
+which can be used to refer to the associated user.
+The user profile essentially becomes the representation of the user,
+even when it's not actively online.
+
+When a user logs in to {kib}, {kib} creates a new profile or associates an
+existing one for the user.
+By using the unique ID of the user profile, {kib} can store user preferences
+separately for each user, which is key to fine-grained level of customisation.
+This unique ID also offers a way to uniquely refer to a user so that
+{kib} can route messages and notifications to the appropriate user,
+regardless of whether they're logged in.
+
+===== Why a new unique ID instead of just the `username`?
+
+The `username` is _not_ unique outside a single realm. Using just the `username`
+is not possible to differentiate two users with the same `username` but from
+different realms.
```

Comment:
A separate unique ID also makes it somewhat easier to support `username` changes. But I leave it out since this possibility hasn't really been explored.